hudi-flink-datasource/hudi-flink/src/main/java/org/apache/hudi/sink/clustering/ClusteringOperator.java

```diff
@@ -100,15 +104,32 @@ public class ClusteringOperator extends TableStreamOperator<ClusteringCommitEven
   private transient AvroToRowDataConverters.AvroToRowDataConverter avroToRowDataConverter;
   private transient HoodieFlinkWriteClient writeClient;
   private transient BulkInsertWriterHelper writerHelper;
-  private transient String instantTime;
 
   private transient BinaryExternalSorter sorter;
   private transient StreamRecordCollector<ClusteringCommitEvent> collector;
   private transient BinaryRowDataSerializer binarySerializer;
 
+  /**
+   * Whether to execute compaction asynchronously.
+   */
+  private final boolean asyncClustering;
+
+  /**
+   * Executor service to execute the clustering task.
+   */
+  private transient NonThrownExecutor executor;
+
+  /**
+   * Flag saying whether the schema is initialized,
+   * the table may be empty when start up, in order to fetch the latest schema,
+   * initializes the schema lazily.
+   */
+  private boolean initialized = false;
```

Comment:
Is it necessary to be lazy initialized ? Based on the fact that we already passed in the `rowType` on construction, only static schema is supported.

Reply:
Modify it, passed schema from `rowType`.